src/chains/ethereum/ethereum/src/blockchain.ts

```diff
@@ -807,7 +807,18 @@ export default class Blockchain extends Emittery<BlockchainTypedEvents> {
         blocksToDelete[0].header.number
       );
     }
-    await this.#database.batch(() => {
+    
+    // notify subscribers that logs have been rolled back
+    await Promise.all(
+      blocksToDelete.map(async block => {
+        const blockNum = block.header.number.toBuffer();
+
+        const logsToDelete = (await this.blockLogs.get(blockNum)).asRemoved();
```

Comment:
Do you have any rough idea as to how much of a speedup this might be?

My intuition is that it wouldn't be huge, as even if there's no explicit caching in the database, I'd imagine that during an automated test run the data would still be resident in the OS disk cache. That said, if decoding is inefficient/expensive, I could imagine it being impactful. At the very least it'd shave off a few ticks of the ol' event loop (likely where the bulk of the savings would come from).